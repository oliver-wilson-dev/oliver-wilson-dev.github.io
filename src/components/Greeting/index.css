.strikethroughJoke {
    text-decoration: #f42109 solid line-through;
    font-style: italic;
}

.paragraphEmoji {
    margin-right: 10px;
    font-size: 30px;
}

.reasonsList {
    list-style-type: none;
    padding-left: unset;
}

.listOrderIndicator {
    margin-right: 10px;
    font-weight: 600;
}