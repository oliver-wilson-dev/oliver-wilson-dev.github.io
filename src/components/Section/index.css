@import "../../styles/transitions.css";

@keyframes fadeIn {
    0% { 
        opacity: 0; 
    }
    100% { 
        opacity: 1; 
    }
}

.section {
    transition: visibility var(--sectionVisibilityTransitionDuration), opacity var(--sectionOpacityTransitionDuration);
    transition-delay: var(--spinnerOpacityTransitionDuration);
    opacity: 0;
    visibility: hidden;
    &.show {
        opacity: 1;
        visibility: visible;
      }
}

.displayBox {
    background: var(--primaryBackground);
    -webkit-box-shadow: 8px 8px 0px var(--primaryBorderColour);
    -moz-box-shadow: 8px 8px 0px var(--primaryBorderColour);
    box-shadow: 8px 8px 0px var(--primaryBorderColour);
    border: 3px solid var(--primaryBorderColour);
    padding: 10px;
    max-height: 250px;
    overflow: hidden;
    position: relative;
}

.displayBoxExpanded {
    max-height: 1500px;
}

.collapseExpand {
    margin: 5px 5px 5px auto;
    position: absolute;
    right: 0px;
    top: 0px;
    font-size: 30px;
    width: 35px;
    border: none;
    background: transparent;
    font-family: inherit;
    font-weight: 900;
    color: inherit;
}

@media all and (min-width: 786px) {
    .landingBlurb {
        max-width: 800px;
    }
}