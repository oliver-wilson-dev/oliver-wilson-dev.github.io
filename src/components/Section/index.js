import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './index.css';
import sharedStyles from '../App/index.css';

class Section extends React.Component {
  constructor(props) {
    super(props);

    this.displayBoxRef = null;
    this.childrenRef = React.createRef();

    this.state = {
      expanded: false,
      expandable: false,
      visible: false
    };

    this.setDisplayBoxRef = this.setDisplayBoxRef.bind(this);

    this.canBeFocusedProgrammatically = -1;
  }


  componentDidMount() {
    this.setState(
      previousState => ({
        ...previousState,
        expandable: this.displayBoxRef.offsetHeight >= 250,
        visible: true
      })
    );
  }

  componentDidUpdate() {
    const { state: { expanded } } = this;

    if (expanded) { this.childrenRef.current.focus(); }
  }

  setDisplayBoxRef(elm) {
    this.displayBoxRef = elm;
  }

  onButtonClick = (e) => {
    e.preventDefault();
    this.setState(previousState => ({
      ...previousState,
      expanded: !previousState.expanded
    }));
  }

  render() {
    const {
      canBeFocusedProgrammatically,
      onButtonClick,
      setDisplayBoxRef,
      props: { title, children },
      state: { expanded, expandable, visible },
    } = this;

    return (
      <div className={cn(sharedStyles.flexColumn, sharedStyles.flexCenter, styles.section, {
        [styles.show]: visible
      })}
      >
        <h2 className={sharedStyles.sectionTitle}>{title}</h2>
        <div
          ref={setDisplayBoxRef}
          className={(cn(styles.landingBlurb, styles.displayBox,
            {
              [styles.displayBoxExpanded]: expanded,
              [styles.blurBottom]: expandable && !expanded
            }))
          }
        >
          {expandable && (
          <button
            className={styles.collapseExpand}
            type="button"
            onClick={onButtonClick}
            aria-expanded={expanded}
            aria-label={expanded ? 'click to minimise this section' : 'click to expand this section to reveal more content'}
          >
            <span>{expanded ? 'âˆ’' : '+'}</span>
          </button>
          )}
          <div
            ref={this.childrenRef}
            tabIndex={canBeFocusedProgrammatically}
            aria-hidden={expandable && !expanded}
          >
            {children}
          </div>
        </div>
      </div>
    );
  }
}

Section.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default Section;
