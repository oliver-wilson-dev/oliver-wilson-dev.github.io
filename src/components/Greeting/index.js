import React from 'react';
import PropTypes from 'prop-types';
import Section from '../Section';
import styles from './index.css';

const Greeting = ({ title, content, openingLine }) => (
  <Section title={title}>
    <p>
      <span className={styles.paragraphEmoji} role="img" aria-hidden>ðŸ‘‹</span>
      {openingLine.start}
      {' '}
      {<span className={styles.strikethroughJoke}>{openingLine.joke}</span>}
      {' '}
      {openingLine.end}
    </p>
    {content.map(paragraph => <p key={paragraph}>{paragraph}</p>)}
  </Section>
);

Greeting.propTypes = {
  content: PropTypes.array.isRequired,
  openingLine: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
};

export default Greeting;
