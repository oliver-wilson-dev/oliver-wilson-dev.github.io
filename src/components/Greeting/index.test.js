import React from 'react';
import { shallow } from 'enzyme';
import Greeting from '.';
import Section from '../Section';

const defaultProps = {
  content: ['test-paragraph', 'test-paragraph'],
  openingLine: {
    start: 'test-start',
    joke: 'test-joke',
    end: 'test-end'
  },
  title: 'test-title',
};

const render = overrideProps => shallow(<Greeting {...defaultProps} {...overrideProps} />);

jest.mock('../Section', () => {
  const Section = () => null;

  return Section;
});

describe('<Greeting/> component', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should match snapshot', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render a Section component and pass it the "title" prop', () => {
    const testTitle = 'test-title';
    expect(render({ title: testTitle }).find(Section).prop('title')).toBe(testTitle);
  });
});
