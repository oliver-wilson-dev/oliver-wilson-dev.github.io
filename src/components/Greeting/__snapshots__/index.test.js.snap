// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting/> component should match snapshot 1`] = `
<Section
  title="test-title"
>
  <p>
    test-start
     
    <span
      className="strikethroughJoke"
    >
      test-joke
    </span>
     
    test-end
  </p>
  <p
    key="test-paragraph"
  >
    test-paragraph
  </p>
  <p
    key="test-paragraph"
  >
    test-paragraph
  </p>
</Section>
`;
